### PR TITLE
[java] LawOfDemeter to support inner builder pattern

### PR DESCRIPTION
**Pull request by gregorriegler**
Adds another conditional to the [LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter) rule to allow for the use of the Inner Builder pattern, like so:
```java
public class Test {
    public void bar() {
        // Inner Builder pattern chained
        final Bar bar = Bar.newBuilder()
            .withFoo("foo")
            .build();
    }
}
```

**Care**: Does not use the _PREFIX_EXCLUSION_PATTERN_ as this would only check against the _baseName_ when it had to check for the _methodName_ in this case.

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 43 violations, 2 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/a914d83f6e1a4e96b3c55be1b8771e56/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/LawOfDemeter.xml [hunk 1]
```diff
@@ -269,6 +269,20 @@ public class Test {
         final FooBuilder fooBuilder = FooBuilder.newBuilder();
         fooBuilder.withBar();
     }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>#1429 False Positive for Law of Demeter</description>
```
Comment (adangel) [inline pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/LawOfDemeter.xml]: To which issue report are you refering? #1429 seems to be unrelated to LawOfDemeterRule...

I'd put a reference to this pull request instead...
